### DISCUSSION TITLE
--load option vs set ::env(NVC_FOREIGN_OBJ) ...

### DISCUSSION THREAD

**Discussion by JimLewis** [nickg/nvc#627] [General]
Hi Nick,
OSVVM CoSim uses the --load option.   The documentation shows it as a RunTime option.

I am running on NVC 1.9-devel.   I understand that now on MSYS2, we no longer need to set the environment variable:
```
set ::env(NVC_FOREIGN_OBJ) VProc.so
```

Erroneously, it was still set under some conditions.   When running in a simulation that did not use VProc.so (and hence one was not available), I noticed that the build process was failing during elaboration.    

So it makes me curious, is load really a RunTime option or is it also intended to be specified during elaboration?

**Comment by nickg** [comment]
This is a bug: it's not necessary to link it at elaboration time any more but I forgot to remove the code that adds `NVC_FOREIGN_OBJ` to the linker arguments.

**Comment by JimLewis** [comment]
@nickg   Thanks.

**Reply by nickg** [reply]
It's fixed in 87cfd8fa0d. There's an MSYS2 package here: https://github.com/nickg/nvc/suites/11075920290/artifacts/563502529

**Comment by JimLewis** [comment]
Thanks.   I also fixed our scripts so that they use the proper options.
